### ISSUE TITLE
Allowing Rigid body to have CSG shape as child node instead of mesh + collision shape

### ISSUE THREAD

**Issue by Toshiwoz** [godotengine/godot#23331]
I am playing around with CSG, and I tried to place a csg shape with collisions enabled under a rigid body node. Of course it doesn't work, but I'm wondering if it wouldn't be possible to modify godot in order to get that information from a CSG shape instead of only from collision shape + mesh instance.

If someone is interested in contributing, you'll have at least one happy(er) user.

I have searched in the issues to see if there's any mention onf this feature request, but found nothing.
Also I am referring to version 3.1, but I guess it was already implied.

**Comment by NewNodeGames** [comment]
I tried and works for me, placing a CSGBox and CollisionShape as child of RigidBody. Don't work with CSGCombiner with collision activated.

**Comment by NewNodeGames** [comment]
Anyway what is the problem of add a collision shape? Is really easy to do. And CSG system is not for use individual primitives, for that you have mesh instance.

**Comment by Toshiwoz** [comment]
Yeah, maybe there's the problem, if you place multiple csg shapes under rigid body it will probably be more difficult to extract a valid collision..

**Comment by bojidar-bg** [comment]
It should be possible to do this, by just not creating a StaticBody in the CSG, but exposing the shape to the parent PhysicsBody.

So, just checking the type of the parent node, and reusing its instance if possible here:
https://github.com/godotengine/godot/blob/720f543caa9cdefbc92273a419d17bb6441b027e/modules/csg/csg_shape.cpp#L46

**Comment by LeonardMeagher2** [comment]
Could optionally do something similar to [CollisionShape](https://github.com/godotengine/godot/blob/0b7f20c7ed85b81540aadcdd3a530d8a28f52b0b/scene/3d/collision_shape.cpp)
